### ISSUE TITLE
Missing focus outline on selected sortable table headers

### ISSUE THREAD

**Issue by dinpatna** [carbon-design-system/carbon-components-angular#2768]
## Detailed description

> Describe in detail the issue you're having.
In a sortable table when the user clicks on a header to sort the data, the focus outline is missing on the clicked header cell.

> Is this a feature request (new component, new icon), a bug, or a general issue?
Bug

> Is this issue related to a specific component?
table component

> What did you expect to happen? What happened instead? What would you like to see changed?
It should show the focus outline for the selected header. It is not showing the outline. 

> What browser are you working in?
Firefox, Chrome and Safari

> What version of the Carbon Design System are you using?
Carbon v11

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?
IBM App Services Portal

## Steps to reproduce the issue

[Carbon Table Component](<https://angular.carbondesignsystem.com/?path=/story/components-table--basic>)

## Additional information
<img width="1024" alt="Screenshot 2024-01-25 at 1 40 08 PM" src="https://github.com/carbon-design-system/carbon-components-angular/assets/9131033/27bcad5e-d4ed-4452-936e-e722405f1577">

**Comment by kush-savani** [comment]
I see, we should have outline on sortable table header. It is supported in [carbon-react](https://react.carbondesignsystem.com/?path=/story/components-datatable-sorting--default) 
<img width="1728" alt="Screenshot 2024-03-01 at 10 25 50 AM" src="https://github.com/carbon-design-system/carbon-components-angular/assets/49565062/6dabe0ed-45c7-4a4b-9918-6284efc1d58d">

I'm picking this bug.
